src/main/java/org/broadinstitute/hellbender/dev/tools/walkers/bqsr/DF_BaseRecalibrator.java

```diff
@@ -0,0 +1,310 @@
+import htsjdk.samtools.SamInputResource;
+import htsjdk.samtools.SamReader;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.ValidationStringency;
+import htsjdk.tribble.Feature;
+import htsjdk.variant.variantcontext.VariantContext;
+import htsjdk.variant.vcf.VCFCodec;
+import org.broadinstitute.hellbender.cmdline.Argument;
+import org.broadinstitute.hellbender.cmdline.ArgumentCollection;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgram;
+import org.broadinstitute.hellbender.cmdline.CommandLineProgramProperties;
+import org.broadinstitute.hellbender.cmdline.programgroups.ReadProgramGroup;
+import org.broadinstitute.hellbender.dev.pipelines.bqsr.BQSR_Dataflow;
+import org.broadinstitute.hellbender.engine.FeatureDataSource;
+import org.broadinstitute.hellbender.engine.FeatureInput;
+import org.broadinstitute.hellbender.engine.dataflow.ReadsSource;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.tools.recalibration.RecalibrationTables;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import java.io.*;
+import java.nio.channels.Channels;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.List;
+
+
+/**
+ * First pass of the base quality score recalibration -- Generates recalibration table based on various covariates
+ * (such as read group, reported quality score, machine cycle, and nucleotide context).
+ * <p>
+ * Dataflow version of BaseRecalibrator.
+ */
+@CommandLineProgramProperties(
+        usage = "First pass of the Base Quality Score Recalibration (BQSR) -- Generates recalibration table based on various user-specified covariates (such as read group, reported quality score, machine cycle, and nucleotide context).",
+        usageShort = "Generates recalibration table",
+        programGroup = ReadProgramGroup.class
+)
+public class DF_BaseRecalibrator extends CommandLineProgram {
+
+    public static final String APP_NAME = "Hellbender";
+
+    // ------------------------------------------
+    // Command-line options
+
+    /**
+     * all the command line arguments for BQSR and its covariates
+     */
+    @ArgumentCollection(doc = "all the command line arguments for BQSR and its covariates")
+    private transient final BaseRecalibrationArgumentCollection BRAC = new BaseRecalibrationArgumentCollection();
+
+    /**
+     * Path to the input (BAM file). Eventually can be local or on GCS.
+     */
+    protected String beforePath = null;
+
+
+    /**
+     * Path to the reference (SAM file). Eventually can be filesystem or gs://
+     */
+    protected String referencePath = null;
+
+    /**
+     * Path to save the output to. Must be local for now.
+     */
+    @Argument(doc = "Path to save the output to. Must be filesystem.", optional = false)
+    protected String outputTablesPath = "df-recaltables.sj";
+
+    // ------------------------------------------
+    // Dataflow-specific options
```

Comment:
You mean the command-line arguments must move? Why can't they be in the walker? This is the pattern that the other walkers follow.